File: B_Sport_Mafia.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    long long n, k;
    cin >> n >> k;
    int left = -1, right = n + 1;
    while (right - left > 1)
    {
        long long m = left + (right - left) / 2;
        if ((n - m) * (n - m + 1) / 2 - m > k)
            left = m;
        else
            right = m;
    }
    cout << right;
}

File: B_Sport_Mafia.py
n,k=map(int,input().split())
left=-1
right=n+1
while right-left>1:
    m=left+(right-left)//2
    if (n-m)*(n-m+1)//2-m>k:
        left=m
    else:
        right=m
print(right)